### ISSUE TITLE
Watch being fired for a different property when the value is changed

### ISSUE THREAD

**Issue by doonetheprune** [vuejs/vue#4090]
Hi

I have come across a strange issue where the watch for a property is called when the property has a default value of an empty object and the value property is changed.  If that makes no sense, I have created a js fiddle to demonstrate it.

[https://jsfiddle.net/rdjjpc7a/90/](https://jsfiddle.net/rdjjpc7a/90/)

**Comment by defcc** [comment]
The same with #4060

**Comment by doonetheprune** [comment]
Hi 

I don't think the two issues are the same as I am not changing the 'nuts' property and the watch is still being fired.

**Comment by defcc** [comment]
@doonetheprune , your nuts prop default function will return a fresh object {},  {} !== {},  it's changed internally.

**Comment by doonetheprune** [comment]
@defcc I agree the default function should return a fresh object however I don't think it should return a new object every time the value property is changed, it should only do it when the component is created.  If you look at fiddle the nuts property should not be affected by the value property.

Please could you reopen as this is not the same as #4060?

**Comment by defcc** [comment]
This is caused by the vue2 render system. 
When input value changed, your app rerenders from top to bottom, the bla component is also affected.

**Comment by doonetheprune** [comment]
@defcc how come then if the nuts property is a string why is the watch not fired?

[https://jsfiddle.net/rdjjpc7a/93/](url)

**Comment by defcc** [comment]
explained [here](https://github.com/vuejs/vue/issues/4060#issuecomment-257325697) , you could take a look.

**Comment by doonetheprune** [comment]
@defcc I have looked at the fiddle in #4060 and can see why they are similar and the underlying issue might be the same. This issue however highlights the problem in a scenario that is a lot more likely to occur. Also reading the comments from @yyx990803  on issue #4060 it makes me really worried you dont plan on fixing this issue. Without trying to be to harsh on vue which I think is great, a basic bug like this means it looses a lot of credibility. If I came across this when I first started using vue I would have stopped using it.
